tsdb/querier.go

```diff
@@ -314,6 +314,18 @@ func PostingsForMatchers(ix IndexReader, ms ...*labels.Matcher) (index.Postings,
 	}
 
 	for _, m := range ms {
+		if m.Type == labels.MatchRegexp && (m.Value == ".*" || m.Value == "^.*$") && len(ms) > 1 {
```

Comment:
Does this work given how we wrap regexes?

Reply:
I believe so. This is value from the promql. We only wrap regexes in `m.re` field.